packages/vm/lib/runTx.ts

```diff
@@ -147,11 +147,7 @@ export default async function runTx(this: VM, opts: RunTxOpts): Promise<RunTxRes
   await state.checkpoint()
 
   // Is it an Access List transaction?
-  if (
-    'transactionType' in opts.tx &&
-    opts.tx.transactionType === 1 &&
-    this._common.isActivatedEIP(2929)
-  ) {
+  if (opts.tx.transactionType !== 0 && this._common.isActivatedEIP(2929)) {
```

Comment:
Just noticed but this does not seem correct anymore, what has EIP 2929 to do with this? I think this was intended to be EIP 2718, to check if typed txs are activated.

Reply:
I guess this was from the "early days" of implementation where you added a lot of EIP checks 😀, guess to be on the safe side in doubt. I think we can (and should) move over to limit these checks to the absolute minimum. Often things are already checked on Common instantiation and are then redundant in other places and therefore not necessary.

In this case here I would say that we can completely drop the check and just go with `if (opts.tx.transactionType !== 0)` since the`EIP-2718` check has already happened in Common if correctly activated with `EIP-2930`. Otherwise the code will throw in the follow-up lines.

Reply:
Will remove the redundant check on 2929 from here.